exchange/exchange_test.go

```diff
@@ -146,12 +146,35 @@ func newRaceCheckingRequest(t *testing.T) *openrtb.BidRequest {
 }
 
 func TestPanicRecovery(t *testing.T) {
+	respStatus := 200
+	respBody := "{\"bid\":false}"
+	server := httptest.NewServer(mockHandler(respStatus, "getBody", respBody))
+	defer server.Close()
+
+	cfg := &config.Configuration{
+		CacheURL: config.Cache{
+			ExpectedTimeMillis: 20,
+		},
+		Adapters: blankAdapterConfig(openrtb_ext.BidderList()),
+	}
+
+	theMetrics := pbsmetrics.NewMetrics(metrics.NewRegistry(), openrtb_ext.BidderList())
```

Comment:
Any reason why we'd not just use the DummyMetricsEngine here for the test?

Reply:
The other panic recovery test I changed above in endpoints/auction_test.go uses the DummyMetricsEngine.  I wanted to exercise a different RecordAdapterPanic() with this test.  The former (auction_test.go) executes the no-op RecordAdapterPanic() in pbsmetrics/config/metrics.go; the latter executes RecordAdapterPanic() in pbsmetrics/go_metrics.go.

Reply:
But, to answer your question, I don't know of a reason why the DummyMetricsEngine wouldn't work here.

Reply:
That makes sense. Can we just leave a comment explaining that though? Because if someone looks at these tests and they see two different MetricsEngine being used, they might get confused and wonder why. In that case a clarifying comment might be helpful. What do you think?

Reply:
Looking more closely at each file, I see that exchange_test.go uses the same metrics engine in each of the four tests in which NewExchange() is called.  OTOH, in auction_test.go, either the dummy metrics engine (twice) or nil (once) is used.  I think people will more likely be comparing tests within the individual files than between these files and so I'm not convinced adding a comment for this one test referring to another test in a file in a separate directory will be particularly beneficial.   Maybe someday we should provide more explanations of the logic behind all the PBS tests, though given how tests are added in incremental fashion, doing so may open many new cans of worms.